rules/repositories.bzl

```diff
@@ -116,30 +133,59 @@ def _rules_ios_bzlmod_dependencies():
             repo = "rules_apple",
             sha256 = "1618fc82e556ebc97ea360b8cacd3365ca3b0e0a85ccb32422468204843e752d",
         )
+        _maybe(
+            rules_apple_api,
+            name = "rules_apple_api",
+            version = "1_0",
+        )
+    elif bazel_version.major == "6" and bazel_version.minor == "1":
+        _maybe(
+            github_repo,
+            name = "build_bazel_rules_swift",
+            project = "bazelbuild",
+            ref = "17e20f7edf27e647f1b45f11ed75d51c17820c3b",
+            repo = "rules_swift",
+            sha256 = "d50c2cb6f1c2c30cf44a8ea60469cd399f7458061169bde76a177b63d6b74330",
+        )
+        _maybe(
+            github_repo,
+            name = "build_bazel_rules_apple",
+            ref = "915ac30a9fa1fd3809599a5ab90fa1c6640fe8dc",
+            project = "bazelbuild",
+            repo = "rules_apple",
+            sha256 = "0204016496a39d5c70247650e098905d129f25347c7e1f019f838ca74252ce2d",
+        )
+        _maybe(
+            rules_apple_api,
+            name = "rules_apple_api",
+            version = "2_0",
+        )
     else:
+        # Bazel 6+ codepath - you can use the 2_x version of rules_apple here if
+        # you'd like to set it. If you want to use a different rules_apple than
+        # we have here - define rules_apple_api - to the value you'd like
```

Comment:
Any Bazel user can already override this to be whatever version they want. They can simply add a rules_apple dependency before rules_ios to override whatever rules_ios provides